gpu-simulator/trace-parser/trace_parser.cc

```diff
@@ -487,3 +429,43 @@ void trace_parser::get_next_threadblock_traces(
     }
   }
 }
+void PipeReader::OpenFile(const std::string &filePath) {
+  if (hasEnding(filePath, ".xz")) {
+    // Use xz command to decompress .xz files
+    command = "xz -dc " + filePath;
+  } else if (hasEnding(filePath, ".traceg")) {
+    // Use cat command for regular trace files
+    command = "cat " + filePath;
+  } else {
+    throw std::runtime_error("Unsupported file type!");
+  }
+
+  // Open the pipe
+  pipe = popen(command.c_str(), "r");
+  if (!pipe) {
+    throw std::runtime_error("Failed to open pipe!");
+  }
+}
+
+bool PipeReader::readLine(std::string &line) {
```

Comment:
We are assuming the trace lines are never over 512 characters.
You may want to consider Linux libc `getline`, which does not appear to be limited by a pre-determined buffer size.
https://www.man7.org/linux/man-pages/man3/getline.3.html

Reply:
I'm actually fine with either way

Reply:
Yea that's why I have an assertion. lol. 
```
assert(line.back() == '\n');
```

I'll take a look.

Reply:
annnnnnd this is exactly the reason why CI is failing. The traces used by the CI does not compress the memory addresses, thus has super long lines.

Let me see if I can use getline.